src/QuantumGravityBridge.sol

```diff
@@ -31,7 +34,8 @@ struct Signature {
 /// (see ./DataRootTuple.sol), with each tuple representing a single data root
 /// in a Celestia block header. Relayed tuples are in the same order as the
 /// block headers.
-contract QuantumGravityBridge is IDAOracle {
+/// @dev DO NOT REMOVE ANY INHERITED CONTRACT! They're essential for upgradability.
```

Comment:
Since IDAOOracle isn't part of the upgradability logic, can it be removed later? If so then maybe document specifically which inherited contracts cant be removed.

Also, is there some way to enforce (potentially imperfectly, but at least as a first line of defence) that they aren't removed in CI? Maybe something like outputting a contract inheritance tree from the compiler and analyzing it.